aws/resource_aws_ssm_document.go

```diff
@@ -266,6 +297,11 @@ func resourceAwsSsmDocumentRead(d *schema.ResourceData, meta interface{}) error
 
 	d.Set("status", doc.Status)
 
+	if v, ok := d.GetOk("attachments"); ok {
+		// The API doesn't currently return attachment information so it has to be set this way
```

Comment:
It appears that the [SSM `GetDocument` API does returns attachment information, but in a different form `AttachmentContent` structure](https://docs.aws.amazon.com/systems-manager/latest/APIReference/API_GetDocument.html#systemsmanager-GetDocument-response-AttachmentsContent). It is generally better to perform the drift detection and support full resource import, but the mismatched structures will make the implementation harder, if not impossible. 🙁 

Since the Create/Update and Read APIs do not share a common API structure, this would have potentially been a case where we could smooth over the user experience in Terraform and support a configuration like the following:

```hcl
attachment {
  name = ""
  url  = ""
}
```

Where on Create/Update the Terraform logic converted those to the requisite `AttachmentSource` objects by automatically inferring the `Key` value from the URL (e.g. `S3FileUrl` if `url` prefix is `s3://`, `SourceUrl` if prefix is `http(s)://` etc). On Read, it would be a simple "passthrough" of the API structure.

However, it appears `AttachmentsSource` `SourceUrl` may support providing the URL to a whole S3 "folder" of files. This concept does not map well into Terraform if the API expands all those into individual `AttachmentContents` objects on Read. Bummer.

---

It may be best in this case then to just rename this attribute directly `attachments_source` (prefer singular for configuration block names) to match the Create/Update API and denote to operators its exact usage.

We can also skip this `d.GetOk()` and `d.Set()` logic in the Read function since it is a root level attribute. Terraform will automatically "passthrough" a given configuration into the Terraform state if it does not have `d.Set()` called on it to overwrite the value during refresh.